src/webclients/components/sdfv/renderer/vscode_renderer.ts

```diff
@@ -206,12 +206,13 @@ export class VSCodeRenderer extends SDFGRenderer {
     }
 
     public showSelectLibraryNodeDialog(callback: CallableFunction): void {
-        const sdfgMetaDict = VSCodeSDFV.getInstance().getMetaDict();
-        if (sdfgMetaDict === null) {
+        if (!VSCodeSDFV.getInstance().getDaemonConnected()) {
```

Comment:
so we cannot see the library nodes if we're offline? aren't they stored in the meta dict?

Reply:
This is a very good point actually: We can see and edit library nodes if offline (i.e. see implementations and expand). What we cannot do is _add_ library nodes to the graph via the SDFG building tools. However, none of the SDFG building tools actually work in offline mode. This was originally simply forgotten, so thank you for pointing this out. I propose a follow-up PR immediately following this one to address this (to keep changes as atomic as possible and deploy property editing offline asap). Thoughts? It could alternatively be hitched into this one.

Reply:
Follow up PR definitely